src/components/spinner/spinner.tsx

```diff
@@ -1,16 +1,32 @@
 import style from '../../styles/style.scss';
-import {h, Component, VNode} from 'preact';
+import {h, Component, VNode, RefObject, createRef} from 'preact';
+import {Text, withText} from 'preact-i18n';
 
 const COMPONENT_NAME = 'Spinner';
 
+const translates = {
+  loading: <Text id="spinner.loading">Loading</Text>
+};
+
 /**
  * Spinner component
  *
  * @class Spinner
  * @example <Spinner />
  * @extends {Component}
  */
+@withText(translates)
 class Spinner extends Component<any, any> {
+  loadingSpan: RefObject<HTMLElement> = createRef<HTMLElement>();
+
+  componentDidMount() {
+    setTimeout(() => {
```

Comment:
what does the setTimeout do ?

Reply:
only if the loading appears more than one second screen reader will announce: "loading" (asked by product)

Reply:
two things
- you need to start the timer from the moment the loading is VISIBLE, not mounted
- i didn't understand the use of ref here. is using a boolean not good enough ?

Reply:
1. I checked it and when the setTimeout is started, loading is already visible.
2. need to update the span with the "loading" message if we want the screen reader to announce it immediately. in order to do it need to save the span in ref and update it.

Reply:
- i would still add a visibility check. i don't know if you can count on it.
- what happens if you don't use a ref ?

Reply:
visibility check was added.